### PR TITLE
fix: cp fix

### PR DESCRIPTION
**Pull request by adil-a**
- SDPA CP doesn't support packed sequences currently. Remove attention mask while performing CP + SDPA.
- Do reductions over the correct meshes for reporting + training loss calculations

### CODE REVIEW

**Review by akoumpa — APPROVED** [review]
LGTM, thank you
